src/modules/settings/handler/handlerSettings.js

```diff
@@ -0,0 +1,96 @@
+import vuexStore from '@/core/store';
+import { mapActions } from 'vuex';
+import { toWei, _ } from 'web3-utils';
+import xss from 'xss';
+import { Toast, ERROR } from '@/modules/toast/handler/handlerToast';
+
+const VALID_FIELDS = [
+  'addressBook',
+  'Errors',
+  'online',
+  'linkQuery',
+  'locale',
+  'stateVersion',
+  'gasLimitWarning',
+  'gasPrice',
+  'gasPriceType',
+  'currentNetwork',
+  'preferredCurrency'
+];
+export default class Settings {
+  constructor() {
+    this.$store = vuexStore;
+    Object.assign(this, mapActions('global', ['setImportedState']));
+  }
+
+  // Receives object from file read in module
+  // Returns a promise so the module can react accordingly
+  importStore(file) {
+    return new Promise((resolve, reject) => {
+      const _this = this;
+      try {
+        const reader = new FileReader();
+        reader.onloadend = evt => {
+          const file = evt.target.result;
+          const obj = JSON.parse(file);
+          const parsedObj = _this._validateImportObject(obj);
+
+          // sets the imported state to the store
+          _this.setImportedState(parsedObj).then(() => {
+            resolve();
+          });
+        };
+        reader.readAsBinaryString(file);
+        reader.onerror(e => {
+          Toast(e.message, {}, ERROR);
+        });
+      } catch (e) {
+        reject(e.message);
+      }
+    });
+  }
+
+  exportStore() {
+    const body = document.body;
+    const time = new Date();
+    const filename = `Store-Export-${time.getTime()}.json`;
+    const newObj = Object.assign({}, this.$store.state.global);
+    const stringifiedGasPrice = newObj.gasPrice.toString();
+    delete newObj['localStore'];
+    delete newObj['gasPrice'];
+    newObj['gasPrice'] = stringifiedGasPrice;
```

Comment:
I dont think this is necessary BN numbers will get converted to strings when you send through JSON.stringify also you dont have to delete localStore